feat(feedback): add word details in final result

dcsiebe opened the pull request:
Store per-word details instead of only the number of errors for feedback.

dcsiebe committed:
feat(feedback): add word details in final result

Store per-word details instead of only the number of errors for feedback.

joggienl commented on docs/api/feedback.md:
```diff
@@ -42,13 +42,49 @@ Content-Type: audio/ogg
+      "recognized": "kenobi",
+      "error": null,
+      "correct": true
+    }
```
We know that we want timing information, can we add it already in the docs?
I was thinking about `recognitionStart` and `recognitionEnd` but it could be simpler maybe...

dcsiebe replied:
I don't think that I can add timing information yet as I don't have a clue about the internal format. I asked about this and am waiting for a reply. As soon as I know more I'll update the docs.

dcsiebe replied:
Timing will not be added in the client results so this isn't relevant any more.

joggienl reviewed (changes requested):
I think we need two changes:

- Timing information
- The table for "request parameters" (line 22 on docs/api/feedback.md) should also state the sentences, not the errors.

dcsiebe committed:
fix(feedback): rename sentences request param

In the request parameters table, update errors to sentences.

dcsiebe committed:
refactor(feedback): make sentences optional

As it's possible to not receive feedback on a feedback recording it
should be possible to store such recordings without the sentence
feedback.

joggienl commented:
Hey Siebe, can you tell us what the status is of this PR?
What needs to be added / removed / changed to complete this?

dcsiebe commented:
Just updated this so the diff is cleaner. I see that I need to add the timing info, after that it's done imo.

dcsiebe committed:
Merge branch 'refs/heads/feedback_api' into feedback_details_pr

dcsiebe commented:
Oh, I'm mistaken, the timing information isn't returned so it doesn't need to be added to the docs. So afaik this is done.
